ecl/eclcc/NEWACTIVITY.rst

```diff
@@ -0,0 +1,537 @@
+Quantile 1 � What is it?
+========================
+This series of blog posts started life as a series of walk-throughs and brainstorming sessions at a team offsite.  The
+series will look at adding a new activity to the system.  The idea is to give an idea of the work
+involved, to highlight the different areas that need changing, and hopefully encourage
+others to add their own activities.  In parallel with the description in this blog there is a series of commits
+to the github repository that correspond to the different stages in adding the activity.  Once the blog is completed,
+the text will also be checked into the source control tree for future reference.
+
+The new activity is going to be a QUANTILE activity, which can be used to find the records that split a dataset into
+equal sized blocks.  Two common uses are to find the median of a set of data (split into 2) or percentiles (split into
+100).  It can also be used to split a dataset for distribution across the nodes in a system.  One hope is that the
+classes used to implement quantile in Thor can also be used to improve the performance of the global sort
+operation.
+
+It may seem fatuous, but the first task in adding any activity to the system is to work out what that activity is going 
+to do!  You can approach this in an iterative manner - starting with a minimal set of functionality and adding options 
+as you think of them � or start with a more complete initial design.  We have used both approaches in the past to add 
+capabilities to the HPCC system, but on this occasion we will be starting from a more complete design that came out of 
+our initial design discussion:
+
+"What are the inputs, options and capabilities that might be useful in a QUANTILE activity?"
+
+The following items came out of the discussion:
```

Comment:
The following items came out of the discussion:
->
Such discussion lead to the following conclusions
"came out of the discussion" doesn't read/sound that good.